compiler/noirc_evaluator/src/ssa/opt/check_u128_mul_overflow.rs

```diff
@@ -60,6 +61,26 @@ impl Function {
     }
 }
 
+// MAX_CONST_VALUE_THAT_DOES_NOT_OVERFLOW_U128_MUL is expected to be [p/U],
+// where U=U128::max() and p is the field modulus.
+//
+// Then x<[p/u]<=p/U, so x*U<p
+static MAX_CONST_VALUE_THAT_DOES_NOT_OVERFLOW_U128_MUL: std::sync::LazyLock<u128> =
+    std::sync::LazyLock::new(|| {
+        let max_const_value_that_does_not_overflow_u128_mul =
+            u128::try_from(FieldElement::modulus() / u128::MAX)
+                .expect("expected max_const_value_that_does_not_overflow to fit into a u128");
+        assert!(
+            BigUint::from(u128::MAX) * max_const_value_that_does_not_overflow_u128_mul
+                < FieldElement::modulus()
+        );
+        max_const_value_that_does_not_overflow_u128_mul
+    });
+
+fn max_const_value_that_does_not_overflow_u128_mul() -> u128 {
```

Comment:
```suggestion
fn max_non_overflowing_const_arg() -> u128 {
```

This function name is pretty wordy atm. Can we shorten this?